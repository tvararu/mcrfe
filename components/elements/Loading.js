import { useState, useEffect } from "react";

export default () => {
  const [dots, setDots] = useState(0);

  useEffect(() => {
    const $interval = setInterval(() => {
      setDots((dots + 1) % 4);
    }, 300);

    return () => {
      clearTimeout($interval);
    };
  }, [dots]);

  return (
    <p className="govuk-body">
      Loading
      {Array(dots)
        .fill(".")
        .join("")}
    </p>
  );
};
