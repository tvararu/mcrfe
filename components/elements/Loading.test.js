import renderer from "react-test-renderer";
import { render, fireEvent, cleanup, act } from "@testing-library/react";
import Loading from "./Loading";

jest.useFakeTimers();

describe("Loading", () => {
  it("renders correctly", () => {
    const component = renderer.create(<Loading />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe("when mounted", () => {
    afterEach(cleanup);

    it("renders with 1 dot", () => {
      const { getByText, container } = render(<Loading />);
      act(() => {
        jest.runOnlyPendingTimers();
      });
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
