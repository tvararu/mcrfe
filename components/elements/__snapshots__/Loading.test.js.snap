// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders correctly 1`] = `
<p
  className="govuk-body"
>
  Loading
  
</p>
`;

exports[`Loading when mounted renders with 1 dot 1`] = `
<p
  class="govuk-body"
>
  Loading
  .
</p>
`;
